### PR TITLE
[BugFix] release rowset id while remove rowset of primary key tablet

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

While finish compaction of primarykey, we don't release the rowset id of expired rowset which cause memory leak.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -2584,6 +2584,7 @@ void TabletUpdates::_remove_unused_rowsets() {
             skipped_rowsets.emplace_back(std::move(rowset));
             continue;
         }
+        StorageEngine::instance()->release_rowset_id(rowset->rowset_id());
         rowset->close();
```
Comment (trueeyu) [inline be/src/storage/tablet_updates.cpp:2588]: release after close and set_need_delete_file()?

Comment (sevev) [reply be/src/storage/tablet_updates.cpp:2587]: > release after close and set_need_delete_file()?

done.

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
@mergify backport branch-2.4

**Comment by sevev** [comment]
@mergify backport branch-2.3

**Comment by sevev** [comment]
@mergify backport branch-2.2

**Pull request merged**